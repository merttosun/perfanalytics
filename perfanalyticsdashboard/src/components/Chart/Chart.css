.chart-wrapper {
  display: flex;
  flex-direction: column;
  width: 400px;
  min-width: 400px;
  justify-content: center;
  border-radius: 6px;
  /* border: 1px solid #4285f4; */
  box-shadow: 0px 0px 11px 0px rgba(158, 152, 158, 1);
  margin: 20px;
}
