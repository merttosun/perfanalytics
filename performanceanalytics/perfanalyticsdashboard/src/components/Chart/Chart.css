.chart-wrapper {
  display: flex;
  flex-direction: column;
  width: 400px;
  min-width: 400px;
  justify-content: center;
  border-radius: 6px;
  /* border: 1px solid #4285f4; */
  box-shadow: 0px 0px 11px 0px rgba(158, 152, 158, 1);
  margin: 20px;
  background: white;
}
.inspect-button {
  height: 25px;
  border: none;
  outline: none;
  background: transparent;
  color: #4285f4;
  font-weight: bold;
  cursor: pointer;
}
.inspect-button:hover {
  background: #4285f4;
  color: white;
  font-size: 14px;
  transition: all 0.4s linear;
  border-bottom-left-radius: 6px;
  border-bottom-right-radius: 6px;
}
