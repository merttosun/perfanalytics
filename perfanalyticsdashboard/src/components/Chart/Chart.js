import React from "react";
import { Line } from "react-chartjs-2";
import "./Chart.css";

export default function Chart(props) {
  const handleInspect = () => {
    props.inspected();
  };
  return (
    <div className="chart-wrapper" data-testid="chart">
      <Line data={props.data} className="line-chart" />
      {props.showInspectButton ? (
        <div className="inspect-button-wrapper" onClick={() => handleInspect()}>
          <img src="./expand.png" />
          <button className="inspect-button" data-testid="inspect-button-id">
            Inspect
          </button>
        </div>
      ) : (
        ""
      )}
    </div>
  );
}

// export default Chart;
