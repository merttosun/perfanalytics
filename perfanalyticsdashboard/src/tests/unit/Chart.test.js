import React, { useState } from "react";
import { render } from "@testing-library/react";
import Chart from "../../components/Chart/Chart";
import { Line } from "react-chartjs-2";
import renderer from "react-test-renderer";

it("qweqweq renders without crashing", () => {
  const data = {
    labels: ["January", "February", "March", "April", "May", "June", "July"],
    datasets: [
      {
        label: "TTFB",
        fill: false,
        lineTension: 0.1,
        backgroundColor: "#4285f4",
        borderColor: "#4285f4",
        borderCapStyle: "butt",
        borderDashOffset: 0.0,
        borderJoinStyle: "miter",
        pointBorderColor: "#4285f4",
        pointBackgroundColor: "#fff",
        pointBorderWidth: 1,
        pointHoverRadius: 5,
        pointHoverBackgroundColor: "red",
        pointHoverBorderColor: "rgba(220,220,220,1)",
        pointHoverBorderWidth: 2,
        pointRadius: 1,
        pointHitRadius: 10,
        data: [165, 259, 80, 81, 356, 1000, 20],
      },
    ],
  };
  const { getByTestId } = render(
    <Chart inspected={() => {}} showInspectButton data={data} />
  );
  const chartContainer = getByTestId("chart");
  expect(chartContainer).toBeInTheDocument();
});
