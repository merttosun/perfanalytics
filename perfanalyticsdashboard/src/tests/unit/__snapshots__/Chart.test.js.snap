// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly date component 1`] = `
<div
  className="chart-wrapper"
  data-testid="chart"
>
  
</div>
`;
